### ISSUE TITLE
Interface for configuring btrfs subvolumes

### ISSUE THREAD

**Issue by Chrysostomus** [calamares/calamares#1580]
**Is your feature request related to a problem? Please describe.**
Users cannot choose their own subvolume layout. 

**Describe the solution you'd like**
In partitioning module, if btrfs is chosen as filesystem, mount point drop down list is replaced with either list with check boxes, or similar interface that allows choosing multiple mount points. Interface similar to how Azure sets tags would also work. The most minimal option would be just a simple and prefilled text input field, that contains a list of mount points separated by spaces. There is label "Btrfs allows mounting partition to multiple mount points using subvolumes". 

Each chosen mount point gets mounted to a subvolume that is being created for this purpose. Subvolume names are @ + last directory name in path of the mount point. For example, /home becomes @home, /var becomes @var, /var/cache becomes @cache. The default subvolumes are chosen from distro configurable list. 

**Describe alternatives you've considered**
Make buttons for subvolumes like lvm has, but this seems like strictly inferior approach. Visualizing subvolumes as sections inside partitions would be misleading. Mount point selection seems the most user friendly option.

**Comment by CicadaSeventeen** [comment]
I also want this feature. But I think it can be easily by editing `/etc/calamares/modules/mount.conf` and not need to change program a lot.
I know nothing about gui programming but I will try to edit it first.

**Comment by oshunluvr** [comment]
At a minimum, we should be able to set the names of the subvolumes at installation. @ and @home are fine, unless like me, you have more than one distro installed to the same BTRFS file system. Going into each install post-installation to edit fstab and grub.cfg, rebooting, re-running update-grub, is a much more lengthy and difficult (as in possibility making typos) than simply setting the subvolume names during installation
 
Don't know if all Calamares installs are set up the same, but when using KDEneon, it's a very simple edit of the file /lib/x86_64-linux-gnu/calamares/mount/main.py to change @ and @home to whatever you want. Of course you have to boot to the live install, then edit the file, then click "Install" but only takes a couple minutes vs. the alternate.

Seems to me that a simple name query when BTRFS is selected wouldn't do too difficult to add to the python script.

**Comment by dalto8** [comment]
> At a minimum, we should be able to set the names of the subvolumes at installation. 

As stated above, the subvolume names are set in `/etc/calamares/modules/mount.conf`.  You can edit that file before starting the installer.

**Comment by CicadaSeventeen** [comment]
> > At a minimum, we should be able to set the names of the subvolumes at installation.
> 
> As stated above, the subvolume names are set in `/etc/calamares/modules/mount.conf`. You can edit that file before starting the installer.

You are right and I have known it for a long time. As maker of livecd, I can set for the subvols of btrfs for calamares.  However it is fine to add a ui for installer so that those who use calamares to install their system can know how to do.
